references/detection/train_tpu.py

```diff
@@ -229,14 +245,42 @@ def do_train(num_cores):
         lr_scheduler.step()
 
 
-
-
 def main():
     FLAGS = parse_flags()
     logger.info("FLAGS: {}".format(FLAGS))
 
-    do_prediction('/coco/train2017/000000566234.jpg', use_cpu=FLAGS.use_cpu)
-    #do_train(FLAGS.num_cores)
+    def mark_step():
+        if not FLAGS.use_cpu:
+            with _bench('mark_step'):
+                xm.mark_step()
+
+    model = torchvision.models.detection.maskrcnn_resnet50_fpn(pretrained=True, is_xla=True)
+    model.eval()
+
+    device = xm.xla_device()
+    if not FLAGS.use_cpu:
+        model = model.to(device)
+
+    if FLAGS.mode == 'eval':
+        dataset_val, _ = get_dataset('coco', 'val', T.ToTensor())
+        dataset_val = torch.utils.data.Subset(dataset_val, range(FLAGS.num_eval))
+
+        data_loader_val = torch.utils.data.DataLoader(
+            dataset_val, batch_size=1, shuffle=False, num_workers=0,
+            collate_fn=utils.collate_fn)
+
+        evaluate(model, data_loader_val, device=device, move_tensors_to_device=False)
+    elif FLAGS.mode == 'test':
+        mark_step()
```

Comment:
this is probably not needed, no operations have been performed at this point iiuc.